maven-failsafe-plugin/src/main/java/org/apache/maven/plugin/failsafe/IntegrationTestMojo.java

```diff
@@ -310,6 +310,22 @@
     @Parameter( property = "failsafe.runOrder", defaultValue = "filesystem" )
     private String runOrder;
 
+    /**
+     * Sets the random seed that will be used to order the tests if {@code failsafe.runOrder} is set to {@code random}.
+     * <br>
+     * <br>
+     * If no seeds are set and {@code failsafe.runOrder} is set to {@code random}, then the seed used will be
+     * outputted (search for "To reproduce ordering use flag -Dfailsafe.seed").
+     * <br>
+     * <br>
+     * To deterministically reproduce any random test order that was run before, simply set the seed to 
+     * be the same value.
+     *
+     * @since 3.0.0-M6
+     */
+    @Parameter( property = "failsafe.seed" )
+    private Long randomSeed;
```

Comment:
Did you take over the changes from an old PR?
We spoke about this parameter that we already have the config parameter named `runOrder`. And so that it should contain the seed too, let's suppose `<runOrder>random1234567890</runOrder>`.
The reason is that we have too many config parameters and this one is not urgent due to we are able to specify the seed in one business parameter `runOrder`.

Reply:
These changes were not taken from an old PR, but I would be happy to take a look and compare the changes if you can point me to the PR.

As for not adding failsafe.seed or surefire.seed, I can change it so that it just parses the numbers after random as the seed instead. Do let me know if that's what is desired here.

Reply:
> Do let me know if that's what is desired here.

Yes this is what I mean. Basically, it is easy to parse the number. You can use regex or the string spliterator.

Reply:
-1, it is two separate params and must be overridable with a system prop easily and *independently* so it is good like that.
You can move it to other parameters or even a map of runOrderParameters to avoid more params but having to merge the steing is bad for end users and not that consistent with the config IMHO.

Reply:
@rmannibucau This was we would have too many config params. Having it in one means that it is about the same business. Here you are setting only run order and so that you say "random" is the same business value with the "seed".
Alternatively it can be done another way with the extensions where this parameter may become an interface. It means that the parameter "runOrder" would become an element what is called complexType in the XML and it may have specific nested elements for the concrete implementation of the interface, and the "seed" would be only for the "random".

Reply:
@Tibor17 fact it creates less parameters is irrelevant - in particular today we have a tons of parameters. Fact it makes it harder to use, integrate and is more error prone for end users is what counts at the end and here having a dedicated variable for the replay is highly relevant IMHO. Also note that having the seed is not always enough to replay a test suite (in particular in random mode) so you can just need to capture the test order and replay it explicitly (a runOrder=fromFile and runOrderFile=/path would be more relevant there).

Reply:
> capture the test order and replay it explicitly (a runOrder=fromFile and runOrderFile=/path would be more relevant there)
I actually have some changes (in a separate local branch) that would accomplish this as well and plan to submit it after this PR is merged. I could still use some help/advice on how to capture the test order (with or without running tests). The goal of my changes would be to empower runOrder to be more customizable and deterministic with seeds, passing in a file, controlling not just test class order but also test method order, etc. 

I have the changes for merging `failsafe.seed` and `surefire.seed` with the `runOrder` parameter ready, but will leave those unpushed for now as I am fine with merging them or keep them separate (as they are now or moving it, etc.).

Reply:
@winglam 
Pls refactor the name `randomSeed` and the property name `failsafe.seed`, and for surefire too.